Polyhedron/demo/Polyhedron/Scene_item.cpp

```diff
@@ -116,276 +116,33 @@ void Scene_item::select(double /*orig_x*/,
 }
 
 // set-up the uniform attributes of the shader programs.
-void Scene_item::attrib_buffers(CGAL::Three::Viewer_interface* viewer, int program_name) const
+void Scene_item::attrib_buffers(CGAL::Three::Viewer_interface* viewer,
+                                int program_name) const
 {
-    GLint is_both_sides = 0;
-    //ModelViewMatrix used for the transformation of the camera.
-    QMatrix4x4 mvp_mat;
-    // ModelView Matrix used for the lighting system
-    QMatrix4x4 mv_mat;
-    // transformation of the manipulated frame
-    QMatrix4x4 f_mat;
-    // used for the picking. Is Identity except while selecting an item.
-    QMatrix4x4 pick_mat;
-    f_mat.setToIdentity();
-    //fills the MVP and MV matrices.
-    GLdouble d_mat[16];
-    viewer->camera()->getModelViewProjectionMatrix(d_mat);
-    //Convert the GLdoubles matrices in GLfloats
-    for (int i=0; i<16; ++i){
-        mvp_mat.data()[i] = GLfloat(d_mat[i]);
+    viewer->attrib_buffers(program_name);
+    QColor c = this->color();
```

Comment:
@maxGimeno: here I have moved most of the code of `Scene_item::attrib_buffers` to a new method `Viewer::attrib_buffers`.